lib/puppet/provider/keystore/keytool.rb

```diff
@@ -0,0 +1,49 @@
+Puppet::Type.type(:keystore).provide(:keytool) do
+  commands :keytool => 'keytool'
+
+  def create
+    generate_keystore
+  end
+
+  def destroy
+    delete_keystore
+  end
+
+  def exists?
+    File.exist?(resource[:keystore])
+  end
+
+  private
+
+  def generate_keystore
+    temp_alias = 'temporary-entry'
+
+    begin
+      keytool(
+        '-genkey',
+        '-storetype', 'pkcs12',
+        '-keystore', resource[:keystore],
+        '-storepass:file', resource[:password_file],
+        '-alias', temp_alias,
+        '-dname', "CN=#{temp_alias}"
+      )
+    rescue Puppet::ExecutionFailure => e
+      Puppet.error("Failed to generate new keystore with temporary entry: #{e}")
```

Comment:
Does error halt execution? It looks like it'll then proceed with attempting to delete the entry, even if it failed.

Reply:
Addressed in the origin PR -- https://github.com/theforeman/puppet-certs/pull/334/files#diff-ab1ff29ab535e60c2e133829981732e7bdc416e5eabf07703b5b32e0ecafda51R18